sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/tuplesketchesAggregates.scala

```diff
@@ -0,0 +1,1895 @@
+ * @param inputAggBufferOffset
+ *   offset for input aggregation buffer
+ */
+case class TupleSketchAggDouble(
+    key: Expression,
+    summary: Expression,
+    lgNomEntries: Expression,
+    mode: Expression,
+    override val mutableAggBufferOffset: Int,
+    override val inputAggBufferOffset: Int)
+    extends TupleSketchAggBase[java.lang.Double, DoubleSummary]
+    with QuaternaryLike[Expression]
+    with SummaryAggregateMode {
+
+  // Constructors
+  def this(key: Expression, summary: Expression) = {
+    this(
+      key,
+      summary,
+      Literal(ThetaSketchUtils.DEFAULT_LG_NOM_LONGS),
+      Literal(ThetaSketchUtils.MODE_SUM),
+      0,
+      0)
+  }
+
+  def this(key: Expression, summary: Expression, lgNomEntries: Expression) = {
+    this(key, summary, lgNomEntries, Literal(ThetaSketchUtils.MODE_SUM), 0, 0)
+  }
+
+  def this(key: Expression, summary: Expression, lgNomEntries: Expression, mode: Expression) = {
+    this(key, summary, lgNomEntries, mode, 0, 0)
+  }
+
+  /**
+   * Override inputTypes to specify key, summary (double/float), lgNomEntries (int), and mode
+   * (string) parameters.
+   */
+  override def inputTypes: Seq[AbstractDataType] =
+    Seq(
+      keyInputTypes,
+      summaryInputType,
+      IntegerType,
+      StringTypeWithCollation(supportsTrimCollation = true))
+
+  /**
+   * Override checkInputDataTypes to validate base inputs (key, summary, lgNomEntries) and mode
+   * parameter.
+   */
+  override def checkInputDataTypes(): TypeCheckResult = {
+    val defaultCheck = checkBaseInputDataTypes()
+    if (defaultCheck.isFailure) return defaultCheck
+
+    checkModeParameter()
+  }
+
+  // Copy constructors required by ImperativeAggregate
+  override def withNewMutableAggBufferOffset(
+      newMutableAggBufferOffset: Int): TupleSketchAggDouble =
+    copy(mutableAggBufferOffset = newMutableAggBufferOffset)
+
+  override def withNewInputAggBufferOffset(newInputAggBufferOffset: Int): TupleSketchAggDouble =
+    copy(inputAggBufferOffset = newInputAggBufferOffset)
+
+  override protected def withNewChildrenInternal(
+      newFirst: Expression,
+      newSecond: Expression,
+      newThird: Expression,
+      newFourth: Expression): TupleSketchAggDouble =
+    copy(key = newFirst, summary = newSecond, lgNomEntries = newThird, mode = newFourth)
+
+  override def first: Expression = key
+  override def second: Expression = summary
+  override def third: Expression = lgNomEntries
+  override def fourth: Expression = mode
+
+  // Overrides for TypedImperativeAggregate
+  override def prettyName: String = "tuple_sketch_agg_double"
+
+  /** Specifies accepted summary input types (double and float). */
+  override protected def summaryInputType: AbstractDataType =
+    TypeCollection(DoubleType, FloatType)
+
+  /**
+   * Creates a DoubleSummaryFactory with the configured aggregation mode.
+   *
+   * @return
+   *   a DoubleSummaryFactory instance configured with the aggregation mode
```

Comment:
This is completely redundant and you can delete the `@return` part entirely :) same elsewhere in the PR.